datasource_permissions.go

```diff
@@ -0,0 +1,88 @@
+package gapi
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+)
+
+// DatasourcePermission has information such as a datasource, user, team, role and permission.
+type DatasourcePermission struct {
+	ID           int64  `json:"id"`
+	DatasourceID int64  `json:"datasourceId"`
+	UserID       int64  `json:"userId"`
+	UserEmail    string `json:"userEmail"`
+	TeamID       int64  `json:"teamId"`
+
+	// Permission levels are
+	// 1 = Query
+	Permission     int64  `json:"permission"`
+	PermissionName string `json:"permissionName"`
+}
+
+type DatasourcePermissionsResponse struct {
+	DatasourceID int64 `json:"datasourceId"`
+	Enabled      bool  `json:"enabled"`
+	Permissions  []*DatasourcePermission
```

Comment:
Is this one missing the JSON mapping?

Reply:
oops. It was auto-mapped due to being public I guess